test/services/GetConsignmentServiceSpec.scala

```diff
@@ -0,0 +1,110 @@
+import org.scalatest.concurrent.ScalaFutures._
+import org.scalatest.time.{Millis, Seconds, Span}
+import sangria.ast.Document
+import sttp.client.HttpError
+import uk.gov.nationalarchives.tdr.GraphQLClient
+import uk.gov.nationalarchives.tdr.GraphQLClient.GraphqlError
+import util.FrontEndTestHelper
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class GetConsignmentServiceSpec extends FrontEndTestHelper {
+
+  implicit val ec: ExecutionContext = ExecutionContext.global
+  implicit val patienceConfig: PatienceConfig = PatienceConfig(timeout = scaled(Span(5, Seconds)), interval = scaled(Span(100, Millis)))
+  case class GraphqlData(data: Option[Data], errors: List[GraphqlError] = Nil)
+
+  private def verifyCaptors(captors: (ArgumentCaptor[Document], ArgumentCaptor[BearerAccessToken], ArgumentCaptor[Option[Variables]]), consignmentId: Long): Unit = {
+    val (documentCaptor, tokenCaptor, variablesCaptor) = captors
+    documentCaptor.getValue should be(gc.document)
+    tokenCaptor.getValue.getValue should be("someAccessToken")
+    variablesCaptor.getValue.isDefined should be(true)
+    variablesCaptor.getValue.get.consignmentId should be(consignmentId)
+  }
+
+  def mockOkResponse(graphQLClient: GraphQLClient[gc.Data, gc.Variables], data: Option[gc.Data], errors: List[GraphqlError]):
+  (ArgumentCaptor[Document], ArgumentCaptor[BearerAccessToken], ArgumentCaptor[Option[Variables]])= {
+    val documentCaptor: ArgumentCaptor[Document] = ArgumentCaptor.forClass(classOf[Document])
+    val tokenCaptor: ArgumentCaptor[BearerAccessToken] = ArgumentCaptor.forClass(classOf[BearerAccessToken])
+    val variablesCaptor: ArgumentCaptor[Option[gc.Variables]] = ArgumentCaptor.forClass(classOf[Option[gc.Variables]])
+
+    val graphqlData = graphQLClient.GraphqlData(data, errors)
+
+    when(graphQLClient.getResult(tokenCaptor.capture(), documentCaptor.capture(), variablesCaptor.capture())).thenReturn(Future(graphqlData))
+    Tuple3(documentCaptor, tokenCaptor, variablesCaptor)
+  }
+
+  def mockFailedResponse(graphQLClient: GraphQLClient[gc.Data, gc.Variables]) = {
+    val documentCaptor: ArgumentCaptor[Document] = ArgumentCaptor.forClass(classOf[Document])
+    val tokenCaptor: ArgumentCaptor[BearerAccessToken] = ArgumentCaptor.forClass(classOf[BearerAccessToken])
+    val variablesCaptor: ArgumentCaptor[Option[gc.Variables]] = ArgumentCaptor.forClass(classOf[Option[gc.Variables]])
+
+    when(graphQLClient.getResult(tokenCaptor.capture(), documentCaptor.capture(), variablesCaptor.capture())).thenReturn(Future.failed(HttpError("")))
+    Tuple3(documentCaptor, tokenCaptor, variablesCaptor)
+  }
+
+  "GetConsignmentService GET" should {
+
+    "Return true when given a valid consignment id" in {
+      val graphQLClient = mock[GraphQLClient[gc.Data, gc.Variables]]
+      val graphQLConfig = mock[GraphQLConfiguration]
+      when(graphQLConfig.getClient[gc.Data, gc.Variables]()).thenReturn(graphQLClient)
+
+      val consignmentId = 999
```

Comment:
How does this consignment ID work with Int rather than Long?

Reply:
In most cases, the Scala compiler can automatically convert an Int to a Long because it's a safe conversion.